x-pack/plugins/security/server/routes/authentication/oidc.ts

```diff
@@ -103,183 +77,122 @@ export function defineOIDCRoutes({
+        query: schema.object(
+          {
+            authenticationResponseURI: schema.maybe(schema.uri()),
+            code: schema.maybe(schema.string()),
+            error: schema.maybe(schema.string()),
+            error_description: schema.maybe(schema.string()),
+            error_uri: schema.maybe(schema.uri()),
+            iss: schema.maybe(schema.uri({ scheme: ['https'] })),
+            login_hint: schema.maybe(schema.string()),
+            target_link_uri: schema.maybe(schema.uri()),
+            state: schema.maybe(schema.string()),
+          },
+          // The client MUST ignore unrecognized response parameters according to
+          // https://openid.net/specs/openid-connect-core-1_0.html#AuthResponseValidation and
+          // https://tools.ietf.org/html/rfc6749#section-4.1.2.
+          { unknowns: 'allow' }
+        ),
+      },
+      options: {
+        access: 'public',
+        excludeFromOAS: true,
+        authRequired: false,
+        tags: [ROUTE_TAG_CAN_REDIRECT, ROUTE_TAG_AUTH_FLOW],
+      },
+    },
+    createLicensedRouteHandler(async (context, request, response) => {
+      // An HTTP GET request with a query parameter named `authenticationResponseURI` that includes URL fragment OpenID
+      // Connect Provider sent during implicit authentication flow to the Kibana own proxy page that extracted that URL
+      // fragment and put it into `authenticationResponseURI` query string parameter for this endpoint. See more details
+      // at https://openid.net/specs/openid-connect-core-1_0.html#ImplicitFlowAuth
+      let loginAttempt: ProviderLoginAttempt | undefined;
+      if (request.query.authenticationResponseURI) {
+        loginAttempt = {
+          type: OIDCLogin.LoginWithImplicitFlow,
+          authenticationResponseURI: request.query.authenticationResponseURI,
+        };
+      } else if (request.query.code || request.query.error) {
+        // An HTTP GET request with a query parameter named `code` (or `error`) as the response to a successful (or
+        // failed) authentication from an OpenID Connect Provider during authorization code authentication flow.
+        // See more details at https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth.
+        loginAttempt = {
+          type: OIDCLogin.LoginWithAuthorizationCodeFlow,
+          //  We pass the path only as we can't be sure of the full URL and Elasticsearch doesn't need it anyway.
+          authenticationResponseURI: request.url.pathname + request.url.search,
+        };
+      } else if (request.query.iss) {
+        // An HTTP GET request with a query parameter named `iss` as part of a 3rd party initiated authentication.
+        // See more details at https://openid.net/specs/openid-connect-core-1_0.html#ThirdPartyInitiatedLogin
+        loginAttempt = {
+          type: OIDCLogin.LoginInitiatedBy3rdParty,
+          iss: request.query.iss,
+          loginHint: request.query.login_hint,
+        };
+      }
```

Comment:
__note:__ same as above, this endpoint should no longer support OP initiated logins (that's why we logged deprecation notice for both deprecated and non-deprecated routes).
```suggestion
      }
```